sonarqube/provider.go

```diff
@@ -119,6 +123,13 @@ type ProviderConfiguration struct {
 
 func configureProvider(d *schema.ResourceData) (interface{}, error) {
 	transport := cleanhttp.DefaultPooledTransport()
+	if proxy, ok := d.Get("http_proxy").(string); ok {
```

Comment:
Under the hood `cleanhttp.DefaultPooledTransport ` uses [https://pkg.go.dev/net/http#ProxyFromEnvironment](https://pkg.go.dev/net/http#ProxyFromEnvironment). 

This property is described as `ProxyFromEnvironment returns the URL of the proxy to use for a given request`

I still don't understand how this is functionally different from setting the proxy directly as SONAR_HOST but thats ok. If i find some time, i'll try write it out myself with an nginx proxy. 

I approved the workflow for the acceptance tests to run, if you can get the tests to run, im ok merging this in

Reply:
Not every proxy is a reverse proxy forwarding requests to the corresponding webserver endpoint. Our usecase is more about routing traffic via a proxy connection to a jumphost in an internal aws vpc. So the proxy doesn't know where to route the traffic. That's done on client side.

> Under the hood `cleanhttp.DefaultPooledTransport ` uses https://pkg.go.dev/net/http#ProxyFromEnvironment.
> 
> This property is described as `ProxyFromEnvironment returns the URL of the proxy to use for a given request`
> 

I know. In large stacks, that call more then one provider, you want to specify which provider is using a proxy.

> I approved the workflow for the acceptance tests to run, if you can get the tests to run, im ok merging this in

Sounds good.